templates/zerver/accounts_home.html

```diff
@@ -75,7 +75,7 @@ <h1 class="get-started">{{ _("Sign up for Zulip") }}</h1>
 
                         {% for backend in page_params.external_authentication_methods %}
                         <div class="login-social">
-                            <form class="form-inline" action="{{ backend.signup_url }}" method="get">
+                            <form class="form-inline" action="{{ backend.signup_url }}?next={{ next }}" method="get">
```

Comment:
This will work fine as long as `backend.signup_url` doesn't already include params - which I believe to be true for all backends at this moment, but not sure if it's certain to hold in the future. But maybe it's okay to do this way, given that I don't think there's a simple way to call arbitrary code here?

Reply:
Yes, I can't think of anything simpler at the moment. Also, since we control the workflow, and it is not user controlled, it is hard to mess this up.

Reply:
We should probably combine in a safe way to avoid a potential land mine with future backend changes - Django and urllib have functions to do so. Feel free to post in #backend if you need help figuring out the best way to do that.

Reply:
This is no longer part of the code. I changed the approach I used to solve the issue.